src/common_components/header/HeaderButton.tsx

```diff
@@ -0,0 +1,58 @@
+"use client";
+import { css } from "@styled-system/css";
+import Image from "next/image";
+import { FC, MouseEventHandler, ReactNode } from "react";
+
+export const HeaderButton: FC<{
+  text: ReactNode;
+  icon: string;
+  clickev: MouseEventHandler<HTMLButtonElement>;
+}> = ({ text, icon, clickev }) => (
```

Comment:
```suggestion
export const HeaderButton: FC<{
  children: ReactNode;
  icon: string;
  clickev: MouseEventHandler<HTMLButtonElement>;
}> = ({ children, icon, clickev }) => (
```
こんな感じにするとHeaderButtonの子要素が自動的にchildrenになるはずなのでこちらのほうが良いかと思います